homepage/models.py

```diff
@@ -12,26 +13,84 @@ class Course(models.Model):
+                                   blank=True, null=True)  # average course load, starts as null
     avg_rating = models.DecimalField(max_digits=6, decimal_places=5,
-                                     validators=[MinValueValidator(1), MaxValueValidator(5)])  # average course rating
-    num_of_raters = models.IntegerField(
-        validators=[MinValueValidator(0)])  # number of raters for course rating and course load
-    num_of_reviewers = models.IntegerField(
-        validators=[MinValueValidator(0)])  # number of reviewers
+                                     validators=[MinValueValidator(1), MaxValueValidator(5)],
+                                     blank=True, null=True)  # average course rating, starts as null
+    num_of_raters = models.IntegerField(validators=[MinValueValidator(0)],
+                                        default=0)  # number of raters for course rating and course load
+    num_of_reviewers = models.IntegerField(validators=[MinValueValidator(0)],
+                                           default=0)  # number of reviewers
 
     def __str__(self):
         return self.name
 
+    def clean(self):
+        rating = self.avg_rating
+        load = self.avg_load
+        raters = self.num_of_raters
+        reviews = self.num_of_reviewers
+
+        if (rating is not None) and (load is None):
+            raise ValidationError(("Can't have rating without load"))
+        if (rating is None) and (load is not None):
+            raise ValidationError(("Can't have load without rating"))
+        if ((rating is None) or (load is None)) and (raters > 0):
+            raise ValidationError(("Can't have raters that didn't rate"))
+        if ((rating is not None) or (load is not None)) and (raters == 0):
+            raise ValidationError(("Can't have ratings without raters"))
+        if ((raters == 0) or (load is None) or (rating is None)) and (reviews > 0):
+            raise ValidationError(("Can not have reviewes without ratings"))
+        if (reviews > raters):
+            raise ValidationError(("Can not have reviewes without ratings"))
+
+    def save(self, *args, **kwargs):
+        try:
+            self.clean()
+        except ValidationError:
+            return
+
+        super().save(*args, **kwargs)
+
+    # updates the course according to review output
+    def upd_course_per_review(self, rating, load, has_content):
+        c_raters = self.num_of_raters
+        c_reviewers = self.num_of_reviewers
+        c_load = self.avg_load
+        c_rating = self.avg_rating
+        if c_load is None:
+            self.avg_load = load
+            self.avg_rating = rating
+            self.num_of_raters = 1
+        else:
+            tot_rating = c_rating * c_raters
+            tot_rating += rating
+            tot_load = c_load * c_raters
+            tot_load += load
+
+            self.avg_rating = tot_rating / (c_raters + 1)

```

Comment:
```suggestion
            self.avg_rating = ((self.avg_rating * self.num_of_raters) + rating) / (self.num_of_raters + 1)

```
And then you can remove the local variables at the beginning of the function, WDYT? 
Same for ```self.avg_load```

Reply:
I actually used local variables on purpose, for a few reasons:
- readability: it is easier to understand what it is when it's called "tot_rating"
- reliability: when I didn't use local variables for current rating and such, the method would sometimes work only partially, and I deduced it was because of the multiple calls to the objects fields, since only after I did the switch it became reliable.

Reply:
Fair enough. 
I've added two comments to this method that IMO can improve readability.

Reply:
I Agree that the local variables add to the readability, but as ori suggested i would change the name of the variables to be more readable

for exmaple change c_raters to course_raters, and tot_rating to total_rating (and also all the similar variables)